src/fastq/fastqread.jl

```diff
@@ -0,0 +1,64 @@
+# FASTQ Read
+# ==========
+#
+# Design discussions in https://github.com/BioJulia/FASTX.jl/pull/35
+
+mutable struct FASTQRead
+    identifier::String
+    description::String
+    sequence::BioSequences.LongDNASeq
+    quality::Vector{UInt8} # in raw PHRED scores, not offset
+end
+
+"""
+    FASTQ.Read(record::FASTQ.Record, offset::Integer=33)
+
+Create a FASTQ read object from a FASTQ.Record.
+The FASTQ.Read has fields: 
+    identifier::String
+    description::String
+    sequence::BioSequences.LongDNASeq
+    quality::Vector{UInt8} # in raw PHRED scores, not offset
+"""
+function Read(record::FASTQ.Record, offset::Integer=33)
+    FASTQRead(
+        identifier(record),
+        description(record),
+        sequence(BioSequences.LongDNASeq, record),
+        quality(record, offset)
+    )
+end
+
+function sequence(read::FASTQ.FASTQRead)
+    read.sequence
+end
+
+function quality(read::FASTQ.FASTQRead)
+    read.quality
+end
+
+function Base.length(read::FASTQ.FASTQRead)
+    length(read.sequence)
+end
+
+function Base.getindex(read::FASTQ.FASTQRead, i::UnitRange{Int})
```

Comment:
Might as well have it be `UnitRange{<:Integer}`

Reply:
fixed in 87cdd30